lib/serialization/base64UrlSpec.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { TypeSpec, createValidationErrorMessage } from "./typeSpec";
+import { SerializationOptions } from "./serializationOptions";
+import { SpecPath } from "./specPath";
+
+/**
+ * A type specification that describes how to validate and serialize a Base64Url encoded ByteArray.
+ */
+const base64UrlSpec: TypeSpec<string, Buffer> = {
+  specType: "Base64Url",
+
+  serialize(propertyPath: SpecPath, value: Buffer, options: SerializationOptions): string {
+    let result: string;
+
+    const anyValue: any = value;
+    if (!anyValue || !anyValue.constructor || typeof anyValue.constructor.isBuffer !== "function" || !anyValue.constructor.isBuffer(value)) {
+      if (options && options.serializationStrictTypeChecking) {
+        throw new Error(createValidationErrorMessage(propertyPath, value, "a Buffer"));
+      }
+      result = anyValue;
+    } else {
+      const base64String: string = value.toString("base64");
+
+      let trimmedResultLength = base64String.length;
+      while ((trimmedResultLength - 1) >= 0 && base64String[trimmedResultLength - 1] === "=") {
```

Comment:
Was this just migrated from the existing serializer or is it new?

Reply:
At a glance, it seems like we should not have to do something like trim the trailing `=` characters from a base64 encoded string.

Reply:
This was migrated from the existing serializer.
We remove the trailing `=` because this is base64URL, which is slightly different than base64.

Reply:
Cool, makes sense.